xds/internal/resolver/cluster_specifier_plugin_test.go

```diff
@@ -39,13 +40,89 @@ import (
 	"google.golang.org/grpc/serviceconfig"
 	"google.golang.org/grpc/xds/internal/balancer/clustermanager"
 	"google.golang.org/grpc/xds/internal/clusterspecifier"
+	xdsresolver "google.golang.org/grpc/xds/internal/resolver"
+	protov2 "google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/anypb"
 	"google.golang.org/protobuf/types/known/wrapperspb"
 
 	v3listenerpb "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
 	v3routepb "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
 )
 
+const (
+	defaultTestTimeout      = 10 * time.Second
+	defaultTestShortTimeout = 100 * time.Microsecond
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+func (s) TestRegister(t *testing.T) {
+	if resolver.Get(xdsresolver.Scheme) == nil {
+		t.Errorf("Scheme %q is not registered", xdsresolver.Scheme)
+	}
+}
+
+// Waits for the resolver to push an update to the fake resolver.ClientConn and
+// verifies that update matches the provided service config.
+//
+// Tests that want to skip verifying the contents of the service config can pass
+// an empty string.
+//
+// Returns the config selector from the state update pushed by the resolver.
+// Tests that don't need the config selector can ignore the return value.
+func verifyUpdateFromResolver(ctx context.Context, t *testing.T, tcc *testutils.ResolverClientConn, wantSC string) iresolver.ConfigSelector {
+	t.Helper()
+
+	var state resolver.State
+	select {
+	case <-ctx.Done():
+		t.Fatalf("Timeout waiting for an update from the resolver: %v", ctx.Err())
+	case state = <-tcc.StateCh:
+		if err := state.ServiceConfig.Err; err != nil {
+			t.Fatalf("Received error in service config: %v", state.ServiceConfig.Err)
+		}
+		if wantSC == "" {
+			break
+		}
+		wantSCParsed := internal.ParseServiceConfig.(func(string) *serviceconfig.ParseResult)(wantSC)
+		if !internal.EqualServiceConfigForTesting(state.ServiceConfig.Config, wantSCParsed.Config) {
+			t.Fatalf("Got service config:\n%s \nWant service config:\n%s", cmp.Diff(nil, state.ServiceConfig.Config), cmp.Diff(nil, wantSCParsed.Config))
+		}
+	}
+	cs := iresolver.GetConfigSelector(state)
+	if cs == nil {
+		t.Fatal("Received nil config selector in update from resolver")
+	}
+	return cs
+}
+
+// buildResolverForTarget builds an xDS resolver for the given target. It
+// returns a fake implementation of the resolver.ClientConn interface that
+// allows inspection of resolver updates, and a function to close the resolver
+// once the test is complete.
+func buildResolverForTarget(t *testing.T, target resolver.Target) (*testutils.ResolverClientConn, resolver.Resolver) {
+	t.Helper()
+
+	builder := resolver.Get(xdsresolver.Scheme)
+	if builder == nil {
+		t.Fatalf("Scheme %q is not registered", xdsresolver.Scheme)
+	}
```

Comment:
I.e. here related to the test comment above

Reply:
Thought about it a little more and got rid of `TestRegister`. Doesn't seem to add much value at all. It if could act as a gate to other tests, i.e failure of this test would prevent other tests from running, then it would be a useful thing to have. Now, if the read from the registry fails, then `TestRegister` would fail and so would all the other tests. So, it makes sense to get rid of it.